src/scanners.rs

```diff
@@ -65,10 +65,23 @@ pub fn is_ascii_whitespace(c: u8) -> bool {
     (c >= 0x09 && c <= 0x0d) || c == b' '
 }
 
+pub fn is_ascii_whitespace_or_carrot(c: u8) -> bool {
+    println!("wh/cr {} {}", c, ((c >= 0x09 && c <= 0x0d) || c == b' ' || c == b'^'));
```

Comment:
I did it this way so that if there were multiple carets `^u^y^y^y` then it would just remove each one so we would see `uyyy` rather than `u^y^y^y`. It's mainly for legacy content.